File: src/main/java/cz/vhromada/catalog/web/validator/ImdbCodeValidator.java
package cz.vhromada.catalog.web.validator;

import java.util.regex.Pattern;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

import cz.vhromada.catalog.utils.Constants;
import cz.vhromada.catalog.web.validator.constraints.ImdbCode;

/**
 * A class represents validator for IMDB code constraint.
 *
 * @author Vladimir Hromada
 */
public class ImdbCodeValidator implements ConstraintValidator<ImdbCode, String> {

    /**
     * IMDB code pattern
     */
    private static final Pattern PATTERN = Pattern.compile("\\d{1,7}");

    /**
     * Minimal IMDB code
     */
    private int minImdb;

    /**
     * Maximal IMDB code
     */
    private int maxImdb;

    @Override
    public void initialize(final ImdbCode imdbCode) {
        minImdb = 1;
        maxImdb = Constants.MAX_IMDB_CODE;
    }

    @Override
    public boolean isValid(final String imdbCode, final ConstraintValidatorContext constraintValidatorContext) {
        if (imdbCode == null) {
            return true;
        }
        if (imdbCode.isEmpty()) {
            return true;
        }
        if (!PATTERN.matcher(imdbCode).matches()) {
            return false;
        }

        final int intValue = Integer.parseInt(imdbCode);
        return intValue >= minImdb && intValue <= maxImdb;
    }

}

File: src/main/kotlin/cz/vhromada/catalog/web/validator/ImdbCodeValidator.kt
package cz.vhromada.catalog.web.validator

import cz.vhromada.catalog.web.validator.constraints.ImdbCode
import cz.vhromada.common.utils.Constants
import java.util.regex.Pattern
import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

/**
 * A class represents validator for IMDB code constraint.
 *
 * @author Vladimir Hromada
 */
class ImdbCodeValidator : ConstraintValidator<ImdbCode, String> {

    override fun isValid(imdbCode: String?, constraintValidatorContext: ConstraintValidatorContext): Boolean {
        if (imdbCode == null || imdbCode.isEmpty()) {
            return true
        }
        if (!PATTERN.matcher(imdbCode).matches()) {
            return false
        }

        return Integer.parseInt(imdbCode) in 1..Constants.MAX_IMDB_CODE
    }

    @Suppress("CheckStyle")
    companion object {

        /**
         * IMDB code pattern
         */
        private val PATTERN = Pattern.compile("\\d{1,7}")

    }

}
